dvc/dvcfile.py

```diff
@@ -0,0 +1,174 @@
+import os
+import re
+import logging
+
+import dvc.prompt as prompt
+
+from voluptuous import MultipleInvalid
+
+from dvc import dependency, output
+from dvc.stage.exceptions import (
+    StageFileBadNameError,
+    StageFileDoesNotExistError,
+    StageFileIsNotDvcFileError,
+    StageFileFormatError,
+    StageFileAlreadyExistsError,
+)
+from dvc.utils import relpath
+from dvc.utils.collections import apply_diff
+from dvc.utils.stage import (
+    parse_stage_for_update,
+    dump_stage_file,
+    parse_stage,
+)
+
+logger = logging.getLogger(__name__)
+
+DVC_FILE = "Dvcfile"
+DVC_FILE_SUFFIX = ".dvc"
+TAG_REGEX = r"^(?P<path>.*)@(?P<tag>[^\\/@:]*)$"
+
+
+class Dvcfile:
+    def __init__(self, repo, path):
+        self.repo = repo
+        self.path, self.tag = self._get_path_tag(path)
+
+    def __repr__(self):
+        return "{}: {}".format(DVC_FILE, self.path)
+
+    @classmethod
+    def is_valid_filename(cls, path):
+        return (
+            path.endswith(DVC_FILE_SUFFIX)
+            or os.path.basename(path) == DVC_FILE
+        )
+
+    @classmethod
+    def is_stage_file(cls, path):
```

Comment:
This kind of naming will be changing on upcoming PRs. For now, stage_file and dvcfile are essentially the same thing and should work.